pkg/reconciler/testing/table.go

```diff
@@ -156,12 +160,42 @@ func (r *TableRow) Test(t *testing.T, factory Factory) {
 			t.Errorf("Unexpected update (-want +got): %s", diff)
 		}
 	}
-	if got, want := len(actions.Updates), len(r.WantUpdates); got > want {
-		for _, extra := range actions.Updates[want:] {
+	if got, want := len(updates), len(r.WantUpdates); got > want {
+		for _, extra := range updates[want:] {
 			t.Errorf("Extra update: %#v", extra)
 		}
 	}
 
+	//TODO(dprotaso)Refactor
+	statusUpdates := filterUpdatesWithSubresource("status", actions.Updates)
+	for i, want := range r.WantStatusUpdates {
+		if i >= len(statusUpdates) {
+			wo := want.GetObject()
+			key := objKey(wo)
+			oldObj, ok := objPrevState[key]
+			if !ok {
+				t.Errorf("Object %s was never created: want: %#v", key, wo)
+				continue
+			}
+			t.Errorf("Missing status update for %s (-want, +prevState): %s", key,
+				cmp.Diff(oldObj, wo, ignoreLastTransitionTime, safeDeployDiff, cmpopts.EquateEmpty()))
+			continue
+		}
+		got := statusUpdates[i].GetObject()
+
+		// Update the object state.
+		objPrevState[objKey(got)] = got
+
+		if diff := cmp.Diff(want.GetObject(), got, ignoreLastTransitionTime, safeDeployDiff, cmpopts.EquateEmpty()); diff != "" {
+			t.Errorf("Unexpected status update (-want +got): %s", diff)
+		}
+	}
+	if got, want := len(statusUpdates), len(r.WantStatusUpdates); got > want {
+		for _, extra := range statusUpdates[want:] {
+			t.Errorf("Extra status update: %#v", extra)
+		}
+	}
+
```

Comment:
We should probably check that `len(updates) + len(statusUpdates) = len(actions.Updates)`

Reply:
See latest commit